analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -297,3 +330,25 @@ internal fun Identifier.toVcsInfo(): VcsInfo {
     val vcsInfo = VcsHost.toVcsInfo("https://$name")
     return vcsInfo.copy(revision = getRevision(version))
 }
+
+/**
+ * Parse the given [lines] generated by the _go mod why_ command and the names of packages that are reported to be
+ * used by the main module.
+ * See https://golang.org/ref/mod#go-mod-why.
+ */
+internal fun parseWhyOutput(lines: Sequence<String>): Set<String> {
+    val usedPackages = mutableSetOf<String>()
+    var currentPackage: String? = null
+
+    lines.forEach { line ->
```

Comment:
The output for a single package is structured as follow

1. First line must be of the form: "# $ID"
2. Followed by 0-N lines which don't start with '(' of the form "$ID"
3. Followed by  a blank line

I believe it would be good to assume that format and make strict checks for it. What do you think?

Reply:
The current approach is indeed pretty tolerant against quirks in the output (although I do not necessarily see this as a disadvantage). In order to apply stricter checks, we would need to carry more state during the iteration over the output lines. There are multiple things that could be wrong, for instance
* multiple blank lines in a series
* blank lines before a line that does not introduce a new package ID
* Multiple lines starting with '#' following each other
* A line starting with '(' not directly after a line with '#'

Should we check all of those?

Reply:
I tried to write in code what I had in mind, but it turned out to be more complex than I expected.
I'm good with leaving it as is.

Reply:
Yes, I also think changing this would be rather complex and would not have that much benefit.